cluster-scope/components/kvm-device-plugin/role.yaml

```diff
@@ -0,0 +1,15 @@
+---
+kind: Role
+apiVersion: authorization.openshift.io/v1
+metadata:
+  name: kvm-device-plugin
+  namespace: kvm-device-plugin
+rules:
+- resources:
+  - securitycontextconstraints
+  apiGroups:
+  - security.openshift.io
+  verbs:
+  - use
+  resourceNames:
+  - privileged
```

Comment:
this looks like it's not going to work since you're using `role` and `rolebindings` but `securitycontextconstraints` is a clusterwide resource

Reply:
Are you sure about this? This is how we're doing it on OpenShift's Prow too (https://github.com/openshift/release/blob/master/clusters/build-clusters/02_cluster/kvm-device-plugin/03-sa-and-role.yaml). I think a Role and RoleBinding does suffice since it only applies to the kvm-device-plugin namespaced here.

Reply:
@LorbusChris [you're right](https://docs.openshift.com/container-platform/4.9/authentication/managing-security-context-constraints.html#role-based-access-to-ssc_configuring-internal-oauth), I guess `scc` are a special case here due to the verb `use` ?  :man_shrugging: 

still...

> Because RBAC is designed to prevent escalation, even project administrators are unable to grant access to an SCC. By default, they are not allowed to use the verb use on SCC resources, including the restricted SCC.

As such can we move the scc roles/rolebindings to `cluster-scope` folder as well? You can create: 

*  `apps/cluster-scope/base/rbac.authorization.k8s.io/roles/kvm-device-plugin/`
* `apps/cluster-scope/base/rbac.authorization.k8s.io/rolebinding/kvm-device-plugin/` 

specify the `kvm-device-plugin` within the kustomization.yaml in those directories and then pull them into the smaug overlay along with the other resources.